### PR TITLE
add ci

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
- [x] build
- [x] setup environment
- [x] add disk test

NOTE: for local images, we could use it to help deploy https://ttl.sh/

### CODE REVIEW

**Comment by Vicente-Cheng** [comment]
Hi @bk201, @guangbochen, 

This PR introduces the basic CI.
Now the CI flow includes four components.

1. Build with PR and push image with changes
2. deploy rancher/longhorn/node-disk-manager, then add one disk on run time
3. Test the added disk is provisioned successfully
4. teardown

I thought that could cover the basic checking with PR changes.

File: .github/workflows/basic-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    needs: tests
+    runs-on:
+      - self-hosted
+      - Linux
+    if: always()
+    steps:
+      - name: Tear Down / Cleanup
+        run: |
+          pushd ndm-vagrant-rancherd
+          vagrant destroy -f node1 node2 node3 node4
```
Comment (bk201) [inline .github/workflows/basic-ci.yaml:66]: We can destroy all VMs by not specifying any VM name.

Comment (Vicente-Cheng) [reply .github/workflows/basic-ci.yaml]: Updated, thanks!

**Review by bk201 — APPROVED** [review]
LGTM, thanks for the great enhancement.

File: .github/workflows/basic-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  deploy:
+    needs: build
+    runs-on:
+      - self-hosted
+      - Linux
+    steps:
+      - name: "Local Deployment (Harvester+Longhorn+Node-Disk-Manager) for testing"
+        run: |
+          rm -rf ndm-vagrant-rancherd
+          git clone https://github.com/bk201/vagrant-rancherd ndm-vagrant-rancherd
```
[the remark this answers is not in the document]
Comment (bk201) [reply .github/workflows/basic-ci.yaml:29]: Transferred: https://github.com/harvester/vagrant-rancherd

File: ci/scripts/deploy_ndm.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    fi
+    echo "harvester-node-disk-manager pods are not ready, sleep 10 seconds."
+    sleep 10
+  done
+}
+
+ensure_longhorn_ready() {
+  # ensure longhorn-manager first
+  while [ true ]; do
+    running_num=$(kubectl get pods -n longhorn-system |grep ^longhorn-manager |grep Running |awk '{print $3}' |wc -l)
```
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply ci/scripts/deploy_ndm.sh]: Thanks for the tips!
updated!

File: ci/scripts/deploy_ndm.sh [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  if ! which $cmd &> /dev/null; then
+    echo 1
+    return
+  fi
+  echo 0
+}
+
+wait_ndm_ready() {
+  while [ true ]; do
+    running_num=$(kubectl get pods -n harvester-system |grep ^harvester-node-disk-manager |grep Running |awk '{print $3}' |wc -l)
```
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply ci/scripts/deploy_ndm.sh]: updated!

File: Dockerfile.dapper [hunk 5]
```diff
@@ -16,7 +16,7 @@ RUN if [ "${ARCH}" = "amd64" ]; then \
 # install openapi-gen
 RUN  go install k8s.io/code-generator/cmd/openapi-gen@v0.20.3
 
-ENV DAPPER_ENV REPO TAG DRONE_TAG CROSS
+ENV DAPPER_ENV REPO TAG DRONE_TAG CROSS BUILD_FORCI
```
[the remark this answers is not in the document]
Comment (Vicente-Cheng) [reply Dockerfile.dapper]: updated!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**